app/src/test/java/ca/klapstein/baudit/data/PasswordTest.java

```diff
@@ -1,36 +1,85 @@
 package ca.klapstein.baudit.data;
 
-import org.junit.After;
-import org.junit.Before;
+
 import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
-public class PasswordTest {
-    @Before
-    public void setUp() {
-    }
+@RunWith(Enclosed.class)
+    public class PasswordTest {
 
-    @After
-    public void tearDown() {
-    }
 
-    @Test
-    public void getPassword() {
-    }
+    @RunWith(Parameterized.class)
+    public class ValidInputs {
+
+        private String input;
+
+        public ValidInputs(String input) {
+            this.input = input;
+        }
+
+        @Parameters
+        public String[] dataValid() {
```

Comment:
I would name `validPasswords` or something similar as it is more descriptive.

`dataValid` is ambiguous.